content/blog/2018-06-07-when-to-use-derived-state.md

```diff
@@ -0,0 +1,331 @@
+---
+title: "When to Use Derived State?"
+author: [bvaughn]
+---
+
+React 16.4 included a [bugfix for getDerivedStateFromProps](/blog/2018/05/23/react-v-16-4.html#bugfix-for-getderivedstatefromprops) that caused some pain in the React community. We apologize for this. We take such changes very seriously. In this case we believe it was the right decision, but we could have done a better job with communication.
+
+[`getDerivedStateFromProps`](/docs/react-component.html#static-getderivedstatefromprops) was introduced in 16.3. At the time, we provided [some examples](/blog/2018/03/27/update-on-async-rendering.html#examples) of how to use the new lifecycle to derive state from props. Our primary goal was to help people migrate from legacy lifecycles to newer ones that are safer to use with the [upcoming async rendering mode](blog/2018/03/01/sneak-peek-beyond-react-16.html).
```

Comment:
Do we need the `getDerivedStateFromProps` link here? There's a lot of links in this paragraph and it makes it a bit harder to scan. I don't think the link is particularly useful because it just gives you a one-sentence description.

If we remove the link I'd suggest starting with English words, e.g.

>The `getDerivedStateFromProps` lifecycle was [...]